### PR TITLE
[catalog] inject a confirmed repository into the local state

### PR DESCRIPTION
**Pull request by das7pad**
Deploying the `dist/` directory to the root of the registry domain picks up the registries URL automatically.

The catalog component requests `registryUI.url() + '/v2/_catalog'` during startup. `registryUI.url()` would return undefined, resulting in just `'/v2/_catalog'`. The Browser is smart enough to complete the url as `window.location.origin + window.location.pathname + '/v2/_catalog'`.

The header of the catalog then says https://github.com/Joxit/docker-registry-ui/blob/a0c88d06f09b9da95e75839353c2e90c962f587f/src/tags/catalog.tag#L22, with `.name` being an alias on `.url`. Resulting in an incomplete header.

Opening any repository would display https://github.com/Joxit/docker-registry-ui/blob/a0c88d06f09b9da95e75839353c2e90c962f587f/src/tags/taglist.tag#L25 resulting in `'Tags of ' + '/' + registryUI.taglist.name`, again incomplete.

The fix is quite simple: once we got a successful response from the _automatically resolved_ server location, we can store the url for the other functions to pickup.

### CODE REVIEW

File: src/tags/catalog.tag [hunk 1]
```diff
@@ -62,6 +62,9 @@ along with this program. If not, see <http://www.gnu.org/licenses/>.
         registryUI.catalog.repositories = [];
       });
       oReq.addEventListener('loadend', function() {
+        if (!registryUI.url()) {
+          registryUI._url = window.location.origin + window.location.pathname.replace(/\/+$/, '')
```
Comment (Joxit) [inline src/tags/catalog.tag]: This is usefull only for the **standard interface** right ? Why don't you put this in `script.js` ?
In this line :
https://github.com/Joxit/docker-registry-ui/blob/36c44b19a3c9be6a80c007b8121a2145472be2a3/src/scripts/script.js#L32-L33

The job is already done for the **static interface** in `static.js`
https://github.com/Joxit/docker-registry-ui/blob/36c44b19a3c9be6a80c007b8121a2145472be2a3/src/scripts/static.js#L18-L24

Comment (das7pad) [reply src/tags/catalog.tag]: Hi,

thank you for your feedback!

I would rather keep the UI in an error state until we get a successful response.

E.g. `Repositories of { registryUI.name() }` filled with the origin/path would not make sense for a UI-only domain.

Comment (Joxit) [reply src/tags/catalog.tag]: Hum ok, but `loadend` event is triggered even if there is an error. This should be done if we get a status code 200?

Comment (das7pad) [reply src/tags/catalog.tag]: Good spot!

I will move this up into the 'load' event into the `this.status == 200` branch.

**Pull request merged**